File: 0121.买卖股票的最佳时机/0121-买卖股票的最佳时机.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) < 2:
            return 0
        profit = 0
        buy,sell = prices[0], None
        for price in prices:
            if price < buy:
                buy = price
            else:
                sell = price
                profit = max(profit, sell-buy)
        return profit

File: 0121.买卖股票的最佳时机/0121-买卖股票的最佳时机.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut profit = 0;
        let mut buy = std::i32::MAX;
        let mut sell = 0;
        for price in prices {
            if price < buy {
                buy = price;
            } else {
                sell = price;
                profit = profit.max(sell-buy);
            }
        }
        
        profit
    }
}